dashboard/components/DevDashboard.vue

```diff
@@ -60,15 +62,26 @@ const campaignList = computed( (): Campaign[] => Object.values( props.campaigns
 const currentCampaign = computed( (): Campaign => campaignList.value.find( ( c: Campaign ) => c.campaign === branchName.value ) );
 const filteredCampaignList = computed( (): Campaign[] => campaignList.value.filter( ( c: Campaign ) => c.campaign !== branchName.value ) );
 
-onMounted( () => {
-	fetch( '/compiled-banners/' )
-		.then( async res => {
-			if ( !res.ok ) {
-				return;
-			}
-			const fileList = await res.text();
-			compileInfo.value = parseCompileInfo( fileList );
+const campaignUrlMap = ref<Record<string, string>>( {} );
+
+onMounted( async () => {
+	try {
+		const bannersResponse = await fetch( '/compiled-banners/' );
+		if ( !bannersResponse.ok ) {
+			return;
+		}
+		const fileList = await bannersResponse.text();
+		compileInfo.value = parseCompileInfo( fileList );
+
+		const pullRequestsResponse = await fetch( `https://api.github.com/repos/wmde/fundraising-banners/pulls?state=open` );
+		const pullRequests = await pullRequestsResponse.json();
+		campaignList.value.forEach( ( campaign: Campaign ) => {
+			const matchingPr = pullRequests.find( ( pr: { title: string; html_url: string } ) => campaign.campaign === pr.title );
```

Comment:
This is a clever way to "fake-type" a JSON result (which usually is `any`). Runtime validation of such results is still a big hassle in TypeScript, even with specialized libraries, so this pattern of "pretending types" like you're doing here (because we *know* the structure will be right) is the appropriate workaround. :+1:

Reply:
Thank you 😊